parallel/runner.go

```diff
@@ -145,3 +150,34 @@ func (r *runner) Cancel() {
 func (r *runner) Errors() map[int]error {
 	return r.errors
 }
+
+// Define the work as done when all consumers are idle for idleThresholdSeconds.
+// Can be run by the producer as a go routine right after starting to produce.
+// CAUTION - Might panic if tasks are taking more than idleThresholdSeconds to be produced.
+func (r *runner) DoneWhenAllIdle(idleThresholdSeconds int) error {
+	for {
+		time.Sleep(time.Duration(idleThresholdSeconds) * time.Second)
+		for i := 0; i < r.maxParallel; i++ {
+			if !r.idle[i] {
+				break
+			}
+
+			idleTimestamp, err := strconv.ParseInt(r.lastActive[i], 10, 64)
+			if err != nil {
+				return errors.New("unexpected idle timestamp on consumer. err: " + err.Error())
+			}
+
+			idleTime := time.Unix(idleTimestamp, 0)
+			now := time.Now()
```

Comment:
This variable definition seem redundant, because `time.Now()` is used aonly once.